_posts/2023-04-19-Announcing-Data-Prepper-2.2.0.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+title:  "Announcing Data Prepper 2.2.0"
+authors:
+- dlv
+date: 2023-04-19 14:30:00 -0500
+categories:
+  - releases
+meta_keywords: Data Prepper 2.1.0
+meta_description: Data Prepper 2.1.0 improves data delivery assurances with end-to-end acknowledgements and an S3 DLQ.
+---
+
+Data Prepper 2.2.0 is now available for [download](https://opensearch.org/downloads.html#data-prepper)!
+This release has a number of changes that help with Data Prepper’s reliability and data delivery assurances.
+
+## S3-based dead-letter queue for OpenSearch
+
+Data Prepper’s `opensearch` sink can now write failed documents to an Amazon S3 dead-letter queue (DLQ) to help with analyzing failures. 
+Prior to this release, the `opensearch` sink would write to a local file only. 
+This required logging into the machine to get these files or creating different infrastructure to export it. 
+Now, the sink can write failed events directly to S3. 
+This will help users that run on serverless infrastructure to avoid maintaining persistent state on serverless machines.
+
+## End-to-end acknowledgements for S3 source
+
+Data Prepper’s `s3` source now supports end-to-end acknowledgements. 
```

Comment:
"Data Prepper's `s3` source now support end-to-end acknowledgments.  

Before E2E, the `s3` source would only acknowledge event delivery with Amazon Simple Queue Service (SQS) after writing all events to a Data Prepper buffer. In cases where Data Prepper was unable to write to OpenSearch, the SQS message would still be acknowledged, and Data Prepper would not read for the object. 

With end-to-end acknowledgments, the `s3` source does not acknowledge completion until all event are sent to an OpenSearch index or the `opensearch` sink's DLQ. If the `s3` source receives no acknowledgment, the SQS message remains in the SQS queue for processing again. 

For Data Prepper 2.2.0, end-to-end acknowledgments are only supported inside the `s3` source since acknowledgments to `s3` are asynchronous. However, we've designed end-to-end acknowledgments so that they could be in other sources. If you would like to see additional sources added for this feature, [create a GitHub issue](https://github.com/opensearch-project/data-prepper/issues/new/choose)."